get-started/part2.md

```diff
@@ -8,125 +8,129 @@ description: Learn how to write, build, and run a simple app -- the Docker way.
+(for a redis database) and rebuild our image as a Docker stack.
 
 - Stack
-- Services
+- Docker Service
 - **Container** (you are here)
 
-## Your new development environment
-
-In the past, if you were to start writing a Python app, your first
-order of business was to install a Python runtime onto your machine. But,
-that creates a situation where the environment on your machine has to be just
-so in order for your app to run as expected; ditto for the server that runs
-your app.
+## Define app image with Dockerfile
 
-With Docker, you can just grab a portable Python runtime as an image, no
-installation necessary. Then, your build can include the base Python image
-right alongside your app code, ensuring that your app, its dependencies, and the
-runtime, all travel together.
+Typically, creating a Python application requires that you install a Python
+runtime, along with the server, and configure it all _perfectly_ on your
+platform for the app to run as expected. With Docker, you simply pull a portable
+Python runtime image from [Docker Hub](https://hub.docker.com/_/python/){:target="_blank"}
+(or [Docker Store](https://store.docker.com/images/python)) and build it with
+your application code (as another image).
 
-These portable images are defined by something called a `Dockerfile`.
+To ensure the integrity and portability of an application, we build images with
+a [Dockerfile](https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/).
+A `Dockerfile` defines the environment inside a container. Access to resources,
+such as networking interfaces and disk drives, is virtualized inside this
+environment. The container is isolated from the rest of your system, so you must
+map ports to the outside world, and be specific about what files you want to
+copy into that environment. Let's start.
 
-## Define a container with `Dockerfile`
+Create an empty directory and change directories into this new directory:
 
-`Dockerfile` will define what goes on in the environment inside your
-container. Access to resources like networking interfaces and disk drives is
-virtualized inside this environment, which is isolated from the rest of your
-system, so you have to map ports to the outside world, and
-be specific about what files you want to "copy in" to that environment. However,
-after doing that, you can expect that the build of your app defined in this
-`Dockerfile` will behave exactly the same wherever it runs.
+```shell
+$ mkdir ~/hellomoby && cd ~/hellomoby
+```
 
-### `Dockerfile`
+Create a file named "Dockerfile" with the following content and save.
 
-Create an empty directory. Change directories (`cd`) into the new directory,
-create a file called `Dockerfile`, copy-and-paste the following content into
-that file, and save it. Take note of the comments that explain each statement in
-your new Dockerfile.
+> Our file refers to `app.py` and `requirements.txt` which we have not yet
+> created. We do this next.
```

Comment:
Is it worth changing the order to make your app files first?

Reply:
Well, requirements.txt is so small, I didn't want it to get lost in the text ...